apa102/src/main/java/com/google/android/things/contrib/driver/apa102/Apa102.java

```diff
@@ -162,13 +163,33 @@ public void setBrightness(int ledBrightness) {
             throw new IllegalArgumentException("Brightness needs to be between 0 and "
                     + MAX_BRIGHTNESS);
         }
-        mLedBrightness = ledBrightness;
+        mLedBrightnessGlobal = ledBrightness;
+        mLedBrightness = null;
     }
 
     /**
-     * Get the current brightness level
+     * Sets the brightness for all LEDs in the strip.
+     * @param ledBrightness The brightness of the LED strip, between 0 and {@link #MAX_BRIGHTNESS}.
+     */
+    public void setIndividualBrightness(int[] ledBrightness) {
+        mLedBrightness = new int[ledBrightness.length];
+        for (int i=0; i<ledBrightness.length; i++) {
+            if (ledBrightness[i] < 0 || ledBrightness[i] > MAX_BRIGHTNESS) {
+                throw new IllegalArgumentException("Brightness needs to be between 0 and "
+                        + MAX_BRIGHTNESS);
+            }
+            mLedBrightness[i] = ledBrightness[i];
+        }
+    }
+
+    /**
+     * Get the current brightness
      */
     public int getBrightness() {
+        return mLedBrightnessGlobal;
+    }
+
+    public int[] getIndividualBrightness() {
         return mLedBrightness;
```

Comment:
this can now return null, you could use `@Nullable`

Reply:
good point, thanks!